### PR TITLE
Dialog: update styles

### LINKED ISSUE #6742: Dialog: Fluent theme

### PR DESCRIPTION
**Pull request by maxwellred**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #6742
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Updating Dialog styles per [new spec](https://app.goabstract.com/projects/82c13930-e212-11e8-a883-231e20d982e7/branches/5842cf9d-814c-44fe-821a-27f367a4482e/files/BAE218DE-4BBA-462C-8C8E-5D3E15116F78/pages/B76CDB68-B3A1-4A14-BEAE-9DAFED8F84D6?filter=changed) from Chris.

Most changes were quite subtle but the largest was changing the large header seen below. I also converted as many colors over to semanticColors as I saw possible.

#### BEFORE (Large Header)

Screen Shot 2019-09-13 at 5 22 37 PM

#### AFTER (Large Header)

Screen Shot 2019-09-13 at 5 22 50 PM

#### BEFORE (Padding at Small breakpoint)
Screen Shot 2019-09-23 at 9 35 08 AM

#### AFTER (Padding at Small breakpoint)
Screen Shot 2019-09-23 at 9 34 56 AM

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10446)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dialog/DialogContent.styles.ts [hunk 1]
```diff
@@ -16,13 +16,16 @@ const GlobalClassNames = {
...
-  const { palette, fonts, effects } = theme;
+  const { palette, fonts, effects, semanticColors } = theme;
 
   const classNames = getGlobalClassNames(GlobalClassNames, theme);
 
   return {
     content: [
-      isLargeHeader && classNames.contentLgHeader,
+      isLargeHeader &&
+        classNames.contentLgHeader && {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Dialog/DialogContent.styles.ts]: why check for contentLgHeader? Won't that always be true?

**Comment by micahgodbolt** [comment]
Dramatic visual changes should be considered possibly breaking (what if they added custom styles to make white text light gray). We need to communicate these changes as much as possible to partner teams. fabricreact alias and Fabric teams channel at least. Max, can you be sure to communicate this out once it gets merged?

**Comment by maxwellred** [comment]
@micahgodbolt I absolutely can, thanks!

File: packages/office-ui-fabric-react/src/components/Dialog/DialogContent.styles.ts (hunk 1 above)
Comment (maxwellred) [reply packages/office-ui-fabric-react/src/components/Dialog/DialogContent.styles.ts]: @dzearing My mistake. My intent was to apply the contentLgHeader classname not use it as a check. I am pushing those changes in now, thanks!

**Comment by betrue-final-final** [comment]
This aligns with the future pattern that Christopher Wymer is building for Fluent.

**Comment by maxwellred** [comment]
I received some quick fixes from Chris and will be implementing them shortly

**Comment by maxwellred** [comment]
Hi @JasonGore, this branch is ready to go with all changes submitted! Let me know if there's anything else you need, thanks!

**Pull request merged**